homeassistant/components/syncthru/config_flow.py

```diff
@@ -0,0 +1,128 @@
+"""Config flow for Samsung SyncThru."""
+
+import re
+from urllib.parse import urlparse
+
+from pysyncthru import SyncThru
+from url_normalize import url_normalize
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_NAME, CONF_URL
+from homeassistant.helpers import aiohttp_client
+
+# pylint: disable=unused-import # for DOMAIN https://github.com/PyCQA/pylint/issues/3202
+from .const import DEFAULT_MODEL, DEFAULT_NAME_TEMPLATE, DOMAIN
+
+
+class SyncThruConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Samsung SyncThru config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle user initiated flow."""
+        if user_input is None:
+            return await self._async_show_form(step_id="user")
+        return await self._async_check_and_create("user", user_input)
+
+    async def async_step_import(self, user_input=None):
+        """Handle import initiated flow."""
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle SSDP initiated flow."""
+        await self.async_set_unique_id(discovery_info[ssdp.ATTR_UPNP_UDN])
+        self._abort_if_unique_id_configured()
+
+        url = discovery_info.get(
+            ssdp.ATTR_UPNP_PRESENTATION_URL,
+            f"http://{urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname}/",
+        )
+        if any(x.data[CONF_URL] == url for x in self._async_current_entries()):
+            return self.async_abort(reason="already_configured")
```

Comment:
If you get here, it means there are existing entries that don't have the right unique ID set. In that case, update the unique ID on that entry before aborting. That way you are improving the config entries.

Reply:
Done.